# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpfe/appshell/AppWindow.cpp
# Commit: bec6b1e2c240
# Full Hash: bec6b1e2c240bd9fcffa14ffb8c7a02fd9c01b29
# Author: seb <seb-firefox@baunz.net>
# Date: 2023-10-27 04:08:13
# Regressor Bug: 1107433
# File Overlap Count: 1
# Description:
#   Bug 1107433 - macOS: Add fallback menu bar for early startup modals. r=mac-reviewers,spohl
#   
#   This adds a native fallback menu for early startup modals (e.g.
#   ProfileManager/ProfileDowngrade) which consists of a "Quit" menu item only,
#   allowing users to quit the app in a regular way (cmd+Q).
# ==============================================================================

diff -r 44e695134648 -r bec6b1e2c240 xpfe/appshell/AppWindow.cpp
--- a/xpfe/appshell/AppWindow.cpp	Thu Oct 26 20:29:54 2023 +0300
+++ b/xpfe/appshell/AppWindow.cpp	Thu Oct 26 16:55:11 2023 +0000
@@ -496,6 +496,14 @@
   nsCOMPtr<nsIWidget> window = mWindow;
   nsCOMPtr<nsIAppWindow> tempRef = this;
 
+#ifdef USE_NATIVE_MENUS
+  // macOS only: For modals created early in startup.
+  // (e.g. ProfileManager/ProfileDowngrade) this creates a fallback menu for
+  // the menu bar which only contains a "Quit" menu item.
+  // This allows the user to quit the application in a regular way with cmd+Q.
+  widget::NativeMenuSupport::CreateNativeMenuBar(mWindow, nullptr);
+#endif
+
   window->SetModal(true);
   mContinueModalLoop = true;
   EnableParent(false);
